pages/[post].tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { useRouter } from "next/router";
+import { dataPost } from "../utils/data-post";
+
+export default function PostOne() {
+  const router = useRouter();
+
+  const postId = router.query.post;
+  if (postId == null) return <p>Loading...</p>;
+
+  const post = dataPost.find((p) => String(p.id) === postId);
```

Comment:
aq tmb da pra desestruturar 
```suggestion
  const {title, imgUrl, text} = dataPost.find((p) => String(p.id) === postId);
```

Reply:
Vale lembrar que find pode retornar null

Reply:
Acho inclusive que eh por isso que ele não fez o destructuring aqui, senao ele nao teria como checar